### ISSUE TITLE
password from passwords.xml not being passed to NAS samba share

### ISSUE THREAD

**Issue by budmannxx** [linuxserver/docker-kodi-headless#104]
Password from _passwords.xml_ not being passed to NAS samba share (or not being passed properly); getting "access denied" error when scanning media and browsing library from webgui.	

* Docker host is: Ubuntu Server 18.04.3 LTS
* Media library is on a FreeNAS SMB share
** library is accessible from the Ubuntu host
* Kodi database is in a separate mariadb container
** kodi-headless container can definitely talk to the MySQL database
** media added to library from other Kodi clients appears in the kodi-headless webgui, and I can see failed scan attempts in the kodi-headless log

_sources.xml_ and _passwords.xml_ have been copied in from working standalone kodi clients.  FreeNAS SMB logs show the correct username coming in, but an incorrect password.  I'm sure the password in _passwords.xml_ is correct.  I have tried multiple usernames (and their respective passwords) in _passwords.xml_, all with the same result.  Relevant log snipptes and config files:

**kodi.log:**
```
2019-10-28 21:18:27.477 T:140417322452736   ERROR: SMBDirectory->GetDirectory: Unable to open directory : 'smb://USERNAME:PASSWORD@192.168.1.197/Movies'
                                            unix_err:'d' error : 'Permission denied'
2019-10-28 21:18:27.477 T:140417322452736   ERROR: GetDirectory - Error getting smb://192.168.1.197/Movies/
2019-10-28 21:18:27.574 T:140417377093376 WARNING: CActiveAE::StateMachine - signal: 22 from port: timer not handled for state: 1
2019-10-28 21:18:29.934 T:140417322452736 WARNING: Previous line repeats 2 times.
2019-10-28 21:18:29.934 T:140417322452736   ERROR: SMBDirectory->GetDirectory: Unable to open directory : 'smb://USERNAME:PASSWORD@192.168.1.197/TV'
                                            unix_err:'d' error : 'Permission denied'
2019-10-28 21:18:29.934 T:140417322452736   ERROR: GetDirectory - Error getting smb://192.168.1.197/TV/
```

**FreeNAS SMB log:**
```
[2019/10/28 16:44:33.526280,  2] ../auth/auth_log.c:610(log_authentication_event_human_readable)
  Auth: [SMB2,(null)] user [WORKGROUP]\[USERNAME] at [Mon, 28 Oct 2019 16:44:33.526245 EDT] with [NTLMv1] status [NT_STATUS_WRONG_PASSWORD] workstation [5F141DFFDAF1] remote host [ipv4:192.168.1.202:48498] mapped to [WORKGROUP]\[USERNAME]. local host [ipv4:192.168.1.197:445]
[2019/10/28 16:44:33.528455,  3] ../auth/auth_log.c:610(log_authentication_event_human_readable)
  Auth: [SMB2,(null)] user []\[] at [Mon, 28 Oct 2019 16:44:33.528446 EDT] with [No-Password] status [NT_STATUS_OK] workstation [] remote host [ipv4:192.168.1.202:48498] became [FREENAS]\[nobody] [S-1-5-21-864646381-810157191-1346284418-501]. local host [ipv4:192.168.1.197:445]
[2019/10/28 16:44:33.529623,  1] ../source3/smbd/service.c:346(create_connection_session_info)
  create_connection_session_info: guest user (from session setup) not permitted to access this share (TV)
[2019/10/28 16:44:33.529647,  1] ../source3/smbd/service.c:529(make_connection_snum)
  create_connection_session_info failed: NT_STATUS_ACCESS_DENIED
```

**passwords.xml** (redacted)**:**
```
<passwords>
    <path>
        <from pathversion="1">smb://192.168.1.197/Movies</from>
        <to pathversion="1">smb://USERNAME:PASSWORD@192.168.1.197/Movies/</to>
    </path>
    <path>
        <from pathversion="1">smb://192.168.1.197/TV</from>
        <to pathversion="1">smb://USERNAME:PASSWORD@192.168.1.197/TV/</to>
    </path>
</passwords>
```
"USERNAME" and "PASSWORD" are correct in the actual file.  The "[No-Password]" entry in the SMB log is what has me thinking there may be a bug here.  Is my _passwords.xml_ misconfigured or is there a bug?  Thx.

**Comment by budmannxx** [comment]
I forgot to add: I was sent to post here, by someone in the Discord channel

**Comment by thezoggy** [comment]
dont you use just put the the passwords in sources.xml ?

**Comment by budmannxx** [comment]
It looks like passwords can be added directly in sources.xml, but I followed the Kodi wiki here: https://kodi.wiki/view/MySQL/Setting_up_Kodi, which says to use the passwords.xml file...

**Comment by thezoggy** [comment]
i've always just done `<path>smb://user:pw@server/path</path>` (you just have to be okay with it leaking that info in your logs)

**Comment by budmannxx** [comment]
I'm trying that now to verify, but the reason I was trying to use passwords.xml was the leakage issue you mention

**Comment by budmannxx** [comment]
I have the same problem when including the share creds in sources.xml.  Pretty much out of ideas at this point.  Maybe I have some config issue with my kodi-headless docker container that is preventing it from seeing the smb shares?  I set the volumes to be mounted when I created the container (technically I'm binding to a directory on the docker host system which is connected to the FreeNAS SMB share).  But when I attach to the Kodi container, I can "ls" the media directory and see all the files there.  So, the Kodi container can definitely access the share...
